src/modules/peek/Peek.UI/MainWindowViewModel.cs

```diff
@@ -63,50 +126,124 @@ public void Initialize(HWND foregroundWindowHandle)
             }
             catch (Exception ex)
             {
-                Logger.LogError("Failed to get File Explorer Items: " + ex.Message);
+                Logger.LogError("Failed to get File Explorer Items.", ex);
             }
 
-            CurrentIndex = 0;
+            _currentIndex = DisplayIndex = 0;
 
-            if (Items != null && Items.Count > 0)
-            {
-                CurrentItem = Items[0];
-            }
+            CurrentItem = (Items != null && Items.Count > 0) ? Items[0] : null;
         }
 
         public void Uninitialize()
         {
-            CurrentIndex = 0;
+            _currentIndex = DisplayIndex = 0;
             CurrentItem = null;
+            _deletedItemIndexes.Clear();
             Items = null;
+            _navigationDirection = NavigationDirection.Forwards;
         }
 
-        public void AttemptPreviousNavigation()
+        public void AttemptPreviousNavigation() => Navigate(NavigationDirection.Backwards);
+
+        public void AttemptNextNavigation() => Navigate(NavigationDirection.Forwards);
+
+        private void Navigate(NavigationDirection direction, bool isAfterDelete = false)
         {
             if (NavigationThrottleTimer.IsEnabled)
             {
                 return;
             }
 
-            NavigationThrottleTimer.Start();
+            if (Items == null || Items.Count == _deletedItemIndexes.Count)
+            {
+                _currentIndex = DisplayIndex = 0;
+                CurrentItem = null;
+                return;
+            }
+
+            _navigationDirection = direction;
+
+            int offset = direction == NavigationDirection.Forwards ? 1 : -1;
+
+            do
+            {
+                _currentIndex = MathHelper.Modulo(_currentIndex + offset, Items.Count);
+            }
+            while (_deletedItemIndexes.Contains(_currentIndex));
+
+            CurrentItem = Items[_currentIndex];
+
+            // If we're navigating forwards after a delete operation, the displayed index does not
+            // change, e.g. "(2/3)" becomes "(2/2)".
+            if (isAfterDelete && direction == NavigationDirection.Forwards)
+            {
+                offset = 0;
+            }
+
+            DisplayIndex = MathHelper.Modulo(DisplayIndex + offset, DisplayItemCount);
 
-            var itemCount = Items?.Count ?? 1;
-            CurrentIndex = MathHelper.Modulo(CurrentIndex - 1, itemCount);
-            CurrentItem = Items?.ElementAtOrDefault(CurrentIndex);
+            NavigationThrottleTimer.Start();
         }
 
-        public void AttemptNextNavigation()
+        /// <summary>
+        /// Sends the current item to the Recycle Bin.
+        /// </summary>
+        public void DeleteItem()
         {
-            if (NavigationThrottleTimer.IsEnabled)
+            if (CurrentItem == null || !IsFilePath(CurrentItem.Path))
```

Comment:
Is this a valid syntax or an exists on disk check? Checking existence on disk is imo a must have. Especially for file on network drives.

Reply:
The `IsFilePath` method checks to ensure the path isn't a folder.

I don't check to see if the file is on a local disk at the moment. Is that what you think should be the case? So we wouldn't allow deletions from a network drive?

Reply:
I mean checking if the file exists in general. (Is not removed by another process.)

Reply:
After rethinking I think as we not spam user with screen messages and won't crash an extra check might be obsolete. What do you think?

Reply:
Yes, I think this would be covered by your proposed change to show a message when the delete operation failed. The user would see something like "Delete failed. File not found". We can therefore avoid doing a pre-delete check for whether the file exists.